app/models/book.rb

```diff
@@ -1,15 +1,23 @@
 class Book < ActiveRecord::Base
   validates_presence_of :title, :published_date, :author, :price
 
-  delegate :sortable_fields, to: :class
+  scope :by_published, -> { order(published_date: :desc) }
 
+  self.per_page = 25
   def self.order_by(field, list_order)
-    list_order ||= "ASC"
-    field ||= "published_date"
-    Book.order("#{field} #{list_order}")
+    if field
+      order(field => list_order)
+    else
+      by_published
+    end
   end
 
   def self.sortable_fields
     [:title, :published_date, :author, :price, :category]
   end
+
+  def self.search(query)
+    search_clauses = sortable_fields.map { |field| "#{field} LIKE :q" }
+    where(search_clauses.join(' OR '), q: "%#{query}%")
+  end
```

Comment:
I saw the discussion in #backend about this. I think this is fine if you're supposed to be avoid using a gem. For real projects, be sure to use a gem.